src/NUnitFramework/framework/Constraints/Operators/ExactCountOperator.cs

```diff
@@ -39,17 +39,26 @@ public class ExactCountOperator : CollectionOperator
         /// <param name="expectedCount">The expected count</param>
         public ExactCountOperator(int expectedCount)
         {
+            // Collection Operators stack on everything
+            // and allow all other ops to stack on them
+            this.left_precedence = 1;
+            this.right_precedence = 10;
+
             this.expectedCount = expectedCount;
         }
 
         /// <summary>
-        /// Returns a constraint that will apply the argument
-        /// to the members of a collection, succeeding if
-        /// none of them succeed.
+        /// Reduce produces a constraint from the operator and 
+        /// any arguments. It takes the arguments from the constraint 
+        /// stack and pushes the resulting constraint on it.
         /// </summary>
-        public override IConstraint ApplyPrefix(IConstraint constraint)
+        /// <param name="stack"></param>
+        public override void Reduce(ConstraintBuilder.ConstraintStack stack)
         {
-            return new ExactCountConstraint(expectedCount, constraint);
+            if (RightContext == null || RightContext is BinaryOperator)
+                stack.Push(new ExactCountConstraint(expectedCount));
+            else
+                stack.Push(new ExactCountConstraint(expectedCount, stack.Pop()));
```

Comment:
@jnm2 Checking for an empty stack works in almost all cases but breaks the case where the expression ends with `.And.Exactly(n).Items`. The test using RightContext is how it's done elsewhere and works for all cases I've found.